tools/src/main/java/org/apache/rocketmq/tools/admin/DefaultMQAdminExtImpl.java

```diff
@@ -418,35 +418,20 @@ public ConsumeStats examineConsumeStats(
     @Override
     public ConsumeStats examineConsumeStats(String consumerGroup,
         String topic) throws RemotingException, MQClientException, InterruptedException, MQBrokerException {
-        TopicRouteData topicRouteData = null;
-        List<String> routeTopics = new ArrayList<>();
-        routeTopics.add(MixAll.getRetryTopic(consumerGroup));
-        if (topic != null) {
-            routeTopics.add(topic);
-            routeTopics.add(KeyBuilder.buildPopRetryTopic(topic, consumerGroup));
-        }
-        for (int i = 0; i < routeTopics.size(); i++) {
-            try {
-                topicRouteData = this.examineTopicRouteInfo(routeTopics.get(i));
-                if (topicRouteData != null) {
-                    break;
-                }
-            } catch (Throwable e) {
-                if (i == routeTopics.size() - 1) {
-                    throw e;
-                }
-            }
-        }
         ConsumeStats result = new ConsumeStats();
-
-        for (BrokerData bd : topicRouteData.getBrokerDatas()) {
-            String addr = bd.selectBrokerAddr();
-            if (addr != null) {
-                ConsumeStats consumeStats = this.mqClientInstance.getMQClientAPIImpl().getConsumeStats(addr, consumerGroup, topic, timeoutMillis * 3);
-                result.getOffsetTable().putAll(consumeStats.getOffsetTable());
-                double value = result.getConsumeTps() + consumeStats.getConsumeTps();
-                result.setConsumeTps(value);
+        for (BrokerData brokerData : this.examineBrokerClusterInfo().getBrokerAddrTable().values()) {
```

Comment:
Scanning all clusters is not a good choice. This way is inefficient and easily leads to timeout, especially since the number of clusters is very large.

Reply:
IMO, old code also scan all clusters, because consumers send heartbeat and create retry topics on all brokers.
Do you have any suggestions for improvement here?

Reply:
> IMO, old code also scan all clusters, because consumers send heartbeat and create retry topics on all brokers. Do you have any suggestions for improvement here?

That's not exactly. If we only have one cluster, these two ways do not have differences. But in production practice, there are many broker clusters registered to the same nameserver cluster. So that your modification will access unnecessary clusters.